tests/ui-cargo/use_crate_prefix_for_self_imports/pass_sibling/src/main.rs

```diff
@@ -0,0 +1,6 @@
+mod foo;
+use foo::Foo;
```

Comment:
What if the order is reversed?

Reply:
Don't lint

Reply:
Will lint

Reply:
That's somewhat unfortunate. I think there should be an easier and quicker way to check for this using resolve (checking if the path is local) that doesn't depend on the item order.

In any event, we should have a test case for this. And if we are order-dependent, I would suggest putting the lint into the `nursery` group for now.